c-api/compat-5.3.c

```diff
@@ -371,14 +376,14 @@ COMPAT53_API int luaL_fileresult (lua_State *L, int stat, const char *fname) {
   int en = errno;  /* calls to Lua API may change this value */
 #if (defined(COMPAT53_HAVE_STRERROR_R) && COMPAT53_HAVE_STRERROR_R) || (defined(COMPAT53_HAVE_STRERROR_S) && COMPAT53_HAVE_STRERROR_S)
   char buf[512] = { 0 };
-#endif // buffer for threadsafe variants of strerror if possible
+#endif /* buffer for threadsafe variants of strerror if possible */
   if (stat) {
     lua_pushboolean(L, 1);
     return 1;
   }
   else {
     lua_pushnil(L);
-#if defined(COMPAT53_HAVE_STRERROR_R)
+#if (defined(COMPAT53_HAVE_STRERROR_R) && COMPAT53_HAVE_STRERROR_R)
```

Comment:
You shouldn't need a `defined` check here.

Reply:
You're right: I don't need it anymore since I forcefully define `COMPAT53_HAVE_STRERROR_R` to be 0. ... But it's no harm, to leave it, right?

Reply:
It's cleaner to take it out. Why would you leave do-nothing conditions lying around in your code?